x/ecocredit/server/msg_server.go

```diff
@@ -525,24 +524,12 @@ func (s serverImpl) getBatchPrecision(ctx types.Context, denom batchDenomT) (uin
 }
 
 // Checks if the given address is in the allowlist of credit class designers
-func (s serverImpl) isDesignerAllowListed(ctx sdk.Context, addr sdk.Address) (bool, error) {
-	var params ecocredit.Params
-	s.paramSpace.GetParamSet(ctx, &params)
-	for _, sAddr := range params.AllowedClassDesigners {
-		allowListedAddr, err := sdk.AccAddressFromBech32(sAddr)
-		if err != nil {
-			return false, err
-		}
-		if addr.Equals(allowListedAddr) {
-			return true, nil
+func (s serverImpl) isDesignerAllowListed(params ecocredit.Params, designer sdk.Address) bool {
+	for _, addr := range params.AllowedClassDesigners {
+		allowListedAddr, _ := sdk.AccAddressFromBech32(addr)
+		if designer.Equals(allowListedAddr) {
+			return true
 		}
 	}
-	return false, nil
-}
-
-// Checks if the allowlist of credit class designers is enabled
-func (s serverImpl) allowlistEnabled(ctx sdk.Context) bool {
```

Comment:
params is now just fetched directly in the `CreateClass` method instead of fetching it twice in both `allowlistEnabled `and `isDesignerAllowlisted`.